### PR TITLE
Feat: add cni virtual device deletion

### PR DESCRIPTION
**Pull request by vicoooo26**
### What type of PR is this?
/kind feature

### What this PR does / why we need it:
This PR allows kk to try to clean the network config created by flannel, cilium and calico when deleting a cluster.

Currently, `kubeadm reset` won't clean the legacy network config created by CNI. If we reinstall a cluster with the previous cluster config after deleting an existing one, new CNI pods may fail to start because of CNI bridge IP address conflicts. 
[k3s upstream](https://github.com/k3s-io/k3s/blob/master/install.sh#L677) now supports flannel cleanup, we need to keep synced with upstream, and add calico and cilium clean up abilities.

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?
None

```release-note

```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: cmd/kk/pkg/bootstrap/os/tasks.go [hunk 1]
```diff
@@ -172,6 +172,15 @@ var (
...
 		"ip link del nodelocaldns",
+		"ip link del cni0",
+		"ip link del flannel.1",
+		"ip link del flannel-v6.1",
+		"ip link del flannel-wg",
+		"ip link del flannel-wg-v6",
+		"ip link del cilium_host",
+		"ip link del cilium_vxlan",
+		"ip -br link show | grep 'cali[a-f0-9]*' | awk -F '@' '{print $1}' | xargs -r -t -n 1 ip link del",
+		"ip netns show 2>/dev/null | grep cni- | xargs -r -t -n 1 ip netns del",
```
Comment (24sama) [inline cmd/kk/pkg/bootstrap/os/tasks.go:183]: Can you paste more information about these 2 commands? Like some documents or screenshots.

**Review by 24sama — COMMENTED** [review]
Thanks for your contribution! And there are some comments, PTAL!

File: cmd/kk/pkg/bootstrap/os/tasks.go (hunk 1 above)
Comment (vicoooo26) [reply cmd/kk/pkg/bootstrap/os/tasks.go:183]: Add some docs about k8s network configuration and description for `delete-cluster` and `delete-node` command. Some explanations may not be accurate, please feel free to revise them.

Comment (24sama) [reply cmd/kk/pkg/bootstrap/os/tasks.go:183]: Thanks!

**Pull request merged**